release-21.2: roachtest: use highmem instances for some tests

erikgrinaker opened the pull request:
Backport:
  * 1/1 commits from "roachtest: use n1-standard for 16-core GCE machines" (#88346)
  * 2/2 commits from "roachtest: use highmem instances for some tests " (#88444)

Please see individual PRs for details.

Resolves #76824.
Resolves #88986.

/cc @cockroachdb/release

Release justification: deflakes tests.

erikgrinaker committed:
roachtest: use n1-standard for 16-core GCE machines

Roachtest used `n1-highcpu` machines at 16 cores and beyond. However,
this causes a memory cliff, because a `n1-standard-8` machine has ~30 GB
memory (3.75 GB per core), but a `n1-highcpu-16` machine only has 14 GB
memory (0.9 GB per core).

This patch makes 16-core machines use `n1-standard` as well, with 60 GB
memory, and only switches to `n1-highcpu` at 32 cores (with 29 GB
memory).

Release note: None

erikgrinaker committed:
roachtest: add `HighMem` option

This patch adds a `HighMem` cluster spec option, which will use machine
types with increased memory. There are no changes to existing machine
types.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88990)
<!-- Reviewable:end -->

erikgrinaker committed:
roachtest: use highmem instances for some tests

The following tests have been seen to OOM occasionally. Since a root
cause fix isn't planned for the near term, this serves to deflake the
tests.

* `tpccbench/nodes=9/cpu=4/multi-region`
* `restore2TB/*`

Release note: None
